e2e/utils/pages/homepage.page.ts

```diff
@@ -57,14 +67,26 @@ abstract class Homepage {
 		await this.#page.setViewportSize(viewportSize);
 	}
 
+	protected async waitForLoginButton(options?: WaitForLocatorOptions): Promise<void> {
+		await this.#page.getByTestId(LOGIN_BUTTON).waitFor(options);
+	}
+
 	protected async waitForHomepageReady(): Promise<void> {
 		await this.goto();
-		await this.#page.getByTestId(LOGIN_BUTTON).waitFor();
+		await this.waitForLoginButton();
 		await this.hideHeroAnimation();
 	}
 
-	protected async waitForTokenSkeletonsInitialization(): Promise<void> {
-		await this.#page.getByTestId(TOKENS_SKELETONS_INITIALIZED).waitFor();
+	protected async waitForTokenSkeletonsInitialization(
+		options?: WaitForLocatorOptions
+	): Promise<void> {
+		await this.#page.getByTestId(TOKENS_SKELETONS_INITIALIZED).waitFor(options);
+	}
+
+	protected async clickMenuItem({ menuItemTestId }: ClickMenuItemParams): Promise<void> {
+		await this.#page.getByTestId(NAVIGATION_MENU_BUTTON).click();
+
```

Comment:
I would suggest to add a `waitFor` here, regarding the popover menu: for example, wait for the popover to appear/mount)

Reply:
Sounds added, added this additional step.